File: app/src/main/java/com/hfad/mymovies/viewmodels/DetailFragmentViewModel.java
package com.hfad.mymovies.viewmodels;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.hfad.mymovies.data.models.Actors;
import com.hfad.mymovies.data.FavouriteMovieRepository;
import com.hfad.mymovies.data.models.FavouriteMovie;
import com.hfad.mymovies.data.models.Movie;
import com.hfad.mymovies.data.models.MovieDiscover;
import com.hfad.mymovies.data.MovieRepository;
import com.hfad.mymovies.data.models.Review;

import java.util.List;
import java.util.Locale;

public class DetailFragmentViewModel extends ViewModel {

    private MovieRepository mMovieRepository;
    private FavouriteMovieRepository mFavouriteMovieRepository;
    private int mMovieId;
    private String language = Locale.getDefault().getLanguage();

    public DetailFragmentViewModel(MovieRepository movieRepository, FavouriteMovieRepository favouriteMovieRepository, int movieId) {
        mMovieRepository = movieRepository;
        mFavouriteMovieRepository = favouriteMovieRepository;
        mMovieId = movieId;
    }

    public LiveData<Movie> getMovie(){
        return mMovieRepository.getMovie(mMovieId, language);
    }

    public void addMovieToFavourite(Movie movie){
        Log.i("ProblemVMadd", "" + movie.getTitle());
        mFavouriteMovieRepository.addMovieToFavourite(movie);
    }

    public void deleteFromFavourite(){
        mFavouriteMovieRepository.deleteFavouriteMovie(
                mFavouriteMovieRepository.getFavouriteMovieById(mMovieId));
    }

    public LiveData<Boolean> isFavourite(){
        return mFavouriteMovieRepository.isFavourite(mMovieId);
    }

    public LiveData<List<Actors>> getActors(){
        return mMovieRepository.getActors(mMovieId, language);
    }

    public LiveData<List<MovieDiscover>> getMovies(){
        return mMovieRepository.getRecommended(1, mMovieId, language);
    }

    public FavouriteMovie getFavouriteMovie(){
        return mFavouriteMovieRepository.getFavouriteMovieById(mMovieId);
    }

    public LiveData<List<Review>> getReviews(){
        return mMovieRepository.getReviews(mMovieId, language);
    }
}

File: app/src/main/java/com/hfad/mymovies/viewmodels/DetailFragmentViewModel.kt
package com.hfad.mymovies.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.hfad.mymovies.data.MovieRepository
import com.hfad.mymovies.data.models.Actors
import com.hfad.mymovies.data.models.Movie
import com.hfad.mymovies.data.models.MovieDiscover
import com.hfad.mymovies.data.models.Review
import java.util.*
import javax.inject.Inject

class DetailFragmentViewModel @Inject constructor(
        private val movieRepository: MovieRepository
//        favouriteMovieRepository: FavouriteMovieRepository,
) : ViewModel() {

    var movieId: Int = 0
    private val language = Locale.getDefault().language
    val movie: LiveData<Movie>
        get() = movieRepository.getMovie(movieId, language)

//    fun addMovieToFavourite(movie: Movie) {
//        mFavouriteMovieRepository.addMovieToFavourite(movie)
//    }

//    fun deleteFromFavourite() {
//        mFavouriteMovieRepository.deleteFavouriteMovie(
//                mFavouriteMovieRepository.getFavouriteMovieById(mMovieId))
//    }

    //    val isFavourite: LiveData<Boolean>
//        get() = mFavouriteMovieRepository.isFavourite(mMovieId)
    val actors: LiveData<List<Actors>>
        get() = movieRepository.getActors(movieId, language)
    val movies: LiveData<MutableList<MovieDiscover>>
        get() = movieRepository.getRecommended(1, movieId, language)

    //    val favouriteMovie: FavouriteMovie
//        get() = mFavouriteMovieRepository.getFavouriteMovieById(mMovieId)
    val reviews: LiveData<List<Review>>
        get() = movieRepository.getReviews(movieId, language)

//    init {
//        this.movieRepository = movieRepository
//        mFavouriteMovieRepository = favouriteMovieRepository
//        mMovieId = movieId
//    }
}